From 04e7b9a3f8b369b1229af9db411c37b2ae779e23 Mon Sep 17 00:00:00 2001
From: Benjamin C Meyer <benjamin.meyer@torchmobile.com>
Date: Wed, 29 Apr 2009 11:15:25 -0400
Subject: [PATCH] When expiring cache files use a QMultiMap, when using a QMap not all
 files are put into the map because often many files (downloaded or
 updated at the same time) will have the same creation QDateTime and
 so only one will go into the QMap who's key is QDateTime.

---
 src/network/access/qnetworkdiskcache.cpp |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/network/access/qnetworkdiskcache.cpp b/src/network/access/qnetworkdiskcache.cpp
index 14a04f1..a44ac2e 100644
--- a/src/network/access/qnetworkdiskcache.cpp
+++ b/src/network/access/qnetworkdiskcache.cpp
@@ -492,21 +492,21 @@ qint64 QNetworkDiskCache::expire()
     QDir::Filters filters = QDir::AllDirs | QDir:: Files | QDir::NoDotAndDotDot;
     QDirIterator it(cacheDirectory(), filters, QDirIterator::Subdirectories);
 
-    QMap<QDateTime, QString> cacheItems;
+    QMultiMap<QDateTime, QString> cacheItems;
     qint64 totalSize = 0;
     while (it.hasNext()) {
         QString path = it.next();
         QFileInfo info = it.fileInfo();
         QString fileName = info.fileName();
         if (fileName.endsWith(CACHE_POSTFIX) && fileName.startsWith(CACHE_PREFIX)) {
-            cacheItems[info.created()] = path;
+            cacheItems.insert(info.created(), path);
             totalSize += info.size();
         }
     }
 
     int removedFiles = 0;
     qint64 goal = (maximumCacheSize() * 9) / 10;
-    QMap<QDateTime, QString>::const_iterator i = cacheItems.constBegin();
+    QMultiMap<QDateTime, QString>::const_iterator i = cacheItems.constBegin();
     while (i != cacheItems.constEnd()) {
         if (totalSize < goal)
             break;
-- 
1.6.0.4

